modules/case-management/src/test/java/com/ibm/cloud/platform_services/case_management/v1/CaseManagementIT.java

```diff
@@ -0,0 +1,367 @@
+            .name("Cloud Object Storage")
+            .type(offeringPayloadTypeModel)
+            .build();
+
+        CreateCaseOptions createCaseOptionsModel = new CreateCaseOptions.Builder()
+            .type("technical")
+            .subject(caseSubject)
+            .description(description)
+            .offering(offeringPayloadModel)
+            .severity(Long.valueOf("4"))
+            .build();
+
+        Response<Case> response = service.createCase(createCaseOptionsModel).execute();
+
+        Case responseObj = response.getResult();
+
+        assertEquals(caseSubject, responseObj.getShortDescription());
+        assertEquals(description, responseObj.getDescription());
+
+        // Storing the new case number for subsequent test cases
+        this.newCaseNumber = responseObj.getNumber();
+
+        System.out.println("New case number: " + responseObj.getNumber());
+    }
+
+    @Test (dependsOnMethods = {"testCreateCase"})
+    public void testGetCases() {
+
+        String[] fields = new String[] {"number", "description"};
+
+        GetCasesOptions getCasesOptionsModel = new GetCasesOptions.Builder()
+            .offset(Long.valueOf("0"))
+            .limit(Long.valueOf("2"))
+            .sort("number")
+            .status(new ArrayList<String>(Arrays.asList("new")))
+            .fields(new ArrayList<String>(Arrays.asList(fields)))
+            .build();
+
+        Response<CaseList> response = service.getCases(getCasesOptionsModel).execute();
+        assertNotNull(response);
+        CaseList responseObj = response.getResult();
+
+        assertTrue(0 < responseObj.getTotalCount());
+        assertTrue(0 < responseObj.getCases().size());
+    }
+
+    @Test (dependsOnMethods = {"testCreateCase"})
+    public void testGetCase() {
+
+        String[] fields = new String[] {"number", "description"};
+
+        GetCaseOptions getCaseOptionsModel = new GetCaseOptions.Builder()
+            .caseNumber(newCaseNumber)
+            .fields(new ArrayList<String>(Arrays.asList(fields)))
+            .build();
+
+        Response<Case> response = service.getCase(getCaseOptionsModel).execute();
+        assertNotNull(response);
+        Case responseObj = response.getResult();
+        assertEquals(newCaseNumber, responseObj.getNumber());
+    }
+
+    @Test (dependsOnMethods = {"testCreateCase"}, expectedExceptions = {BadRequestException.class})
+    public void testGetCasewithInvalidField() {
+
+        String[] fields = new String[] {"number", "description", "invalid_field"};
+
+        GetCaseOptions getCaseOptionsModel = new GetCaseOptions.Builder()
+            .caseNumber(newCaseNumber)
+            .fields(new ArrayList<String>(Arrays.asList(fields)))
+            .build();
+
+        Response<Case> response = service.getCase(getCaseOptionsModel).execute();
+        assertNotNull(response);
+        Case responseObj = response.getResult();
+        assertEquals(newCaseNumber, responseObj.getNumber());
```

Comment:
Using `invalid_field` will return a `400` error instead of a case resposne.

Reply:
It returned a BadRequestException exception.

Reply:
Yeah 400 error is a `Bad Request`. So we should assert something different here.

Reply:
Talked to Huy offline and we are on the same page now.